frontend/src/metabase/common/components/DataPicker/components/TablePicker.tsx

```diff
@@ -78,31 +79,32 @@ export const TablePicker = ({ databaseId, value, onChange }: Props) => {
   );
 
   const handleFolderSelect = useCallback(
-    ({ folder }: { folder: NotebookDataPickerFolderItem }) => {
+    ({ folder }: { folder: DataPickerFolderItem }) => {
       if (folder.model === "database") {
         if (dbId === folder.id) {
           setSchemaName(schemas?.length === 1 ? schemas[0] : undefined);
         } else {
           setDbId(folder.id);
           setSchemaName(undefined);
         }
-        setTableId(undefined);
       }
 
       if (folder.model === "schema") {
         setSchemaName(folder.id);
-        setTableId(undefined);
       }
+
+      setTableId(undefined);
+      onItemSelect(folder);
```

Comment:
All other *Picker components call `onItemSelect` for both value-items and folder-items - `TablePicker` will need to do the same now.